soccer/radio/Radio.hpp

```diff
@@ -23,15 +26,31 @@ class Radio {
 
     int channel() const { return _channel; }
 
-    const std::vector<Packet::RadioRx>& reversePackets() const {
-        return _reversePackets;
+    bool hasReversePackets() {
+        std::lock_guard<std::mutex> lock(_reverse_packets_mutex);
+        return _reversePackets.size();
     }
 
-    std::vector<Packet::RadioRx>& reversePackets() { return _reversePackets; }
+    inline const Packet::RadioRx popReversePacket() {
```

Comment:
Matt mentioned this on my use of inline, I figure I'd copy it here.

"I'd encourage you to actually measure if this inline keyword is necessary. My guess is it's not, but as with all performance things, it doesn't matter unless you measure it. (If you can't be bothered to check, then you don't need the optimization)"

As an aside, while I was googling around for inline stuff, I found this faq from the ioc. The last 3 q&a's are interesting in there's of where to put the inline decorator. I don't think it matters in this case, but it's an interesting thought none the less.
https://isocpp.org/wiki/faq/inline-functions#inline-member-fns

Reply:
The inline wasn't for performance. I'm aware that it's basically ignored by modern compilers. This is a (non-template/constexpr) method definition in a header, which does in fact need inline. If there were two things that had separately included Radio.hpp before this commit, it would have been a link-time error.

Reply:
Actually, my bad here. Forgot that C++ automatically inserts an inline for class members defined in the class declaration, so this is really redundant. I'll take it out.